contracts/Libraries/NTransferUtilV1.sol

```
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity >=0.4.22 <0.9.0;
+import "openzeppelin-solidity/contracts/utils/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+
+library NTransferUtilV1 {
+  using SafeMath for uint256;
+
+  function safeTransfer(
+    IERC20 malicious,
+    address recipient,
+    uint256 amount
+  ) public {
+    if (amount == 0) {
+      return;
```

Instead of silently returning, this function show throw an error.